ext/tiny_tds/client.c

```diff
@@ -326,8 +326,6 @@ static VALUE rb_tinytds_connect(VALUE self, VALUE opts) {
     dbsetlapp(cwrap->login, StringValueCStr(app));
   if (!NIL_P(ltimeout))
     dbsetlogintime(NUM2INT(ltimeout));
-  if (!NIL_P(timeout))
-    dbsettime(NUM2INT(timeout));
```

Comment:
I've fixed this behaviour in FreeTDS in the meantime: https://github.com/FreeTDS/freetds/pull/107. So from now it'll be possible to just simply call `dbsettime(timeout)` each time.

We should still use this original solution though to be compatible with existing versions.